api/auth.proto

```diff
@@ -44,3 +46,78 @@ message AuthAction {
   // List of rules
   repeated Rule rule = 2;
 }
+
+service Authorization {
+	rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+
+	// Address of the downstream source of the request.
+	Address source = 1;
+
+	// Address of the upstream destination of the request.
+	Address destination = 2; 
+
+	// If the request is an HTTP request, the parameters
+	// of the HTTP request.
+	HTTPRequest http = 3;
+
+	// If the request came over a secure TLS connection, information about the connection and 
+	// downstream client.
+	TLSConnection tls = 4;
+}
+
+message HTTPRequest {
+
+	// HTTP Version of the request, e.g. "1.1"
+	string version = 1;
+
+	// Method, e.g. "GET"
+	string method = 2;
+
+	// The value of the Host: header
+	string host = 3;
+
+	// The request URI path.
+	string path = 4;
+
+	// Additional HTTP headers and their values.
+	map<string, string> headers = 5;
+}
+
+message TLSConnection {
```

Comment:
Can we potentially share TLSProperties from here? https://github.com/goaway/data-plane-api/blob/11d0c4eac88390eb75ddd2e29639b74aaace7077/api/filter/accesslog.proto

(In general just trying to remove duplication wherever we can)

Reply:
+1

Reply:
Yeah, they could be merged.